src/storage/query/QueryBoundProcessor.cpp

```diff
@@ -197,8 +205,7 @@ kvstore::ResultCode QueryBoundProcessor::processVertex(PartitionID partId, Verte
     }
 
     if (onlyVertexProps_) {
-        std::lock_guard<std::mutex> lg(this->lock_);
```

Comment:
Well done.  I want to how much the perf being improved without the lock?

Reply:
much smaller than the cache optimization.

Reply:
storaged:
![image](https://user-images.githubusercontent.com/6092236/89253025-00f3eb00-d64e-11ea-97d6-ac41562b12b4.png)

graphd:
![image](https://user-images.githubusercontent.com/6092236/89253043-10733400-d64e-11ea-979d-bdfccb5553f9.png)

What else I can to to help improve the performance.

Reply:
Need to write a custom prefix bloom filter.

Reply:
For GoExecutor,  current logic is not reasonable when yield dest properties.  Actually we don't need VertexHolder. 

When the last step getNeighbors invoked,  the edge props have been fetched,  so we could combine them directly after fetch dest points props.  VertexHolder is needless.

Reply:
> Need to write a custom prefix bloom filter.

The problem about prefix_extractor is we store some system information (such as parts hold by one rocksdb instance). The format is not suitable for prefix_extractor.

Reply:
More further,  the InterimResult has big performance issue.  It will introduce extra encode/decode for the response. 
If we fix the problem,  "Go .... yield dstId | fetch props $- ...."  will be a better choice for you.

Currently,  you could use two queries "Go ..." and "Fetch props" on client separately which will acquire better performance.

Reply:
> For GoExecutor, current logic is not reasonable when yield dest properties. Actually we don't need VertexHolder.
> 
> When the last step getNeighbors invoked, the edge props have been fetched, so we could combine them directly after fetch dest points props. VertexHolder is needless.

VertexHolder is needed currently.

Reply:
Anyway,  the pr totally looks good to me.  Let's open another one for it.